waku/node/rest/filter/handlers.nim

```diff
@@ -50,106 +52,188 @@ func decodeRequestBody[T](contentBody: Option[ContentBody]) : Result[T, RestApiR
                                       peerOpt.get())
 
-    if not await subFut.withTimeout(futTimeoutForSubscriptionProcessing):
-      error "Failed to subscribe to contentFilters do to timeout!"
-      return RestApiResponse.internalServerError("Failed to subscribe to contentFilters")
+  if not await subFut.withTimeout(futTimeoutForSubscriptionProcessing):
+    error "Failed to subscribe to contentFilters do to timeout!"
+    return makeRestResponse(req.requestId, FilterSubscribeError.serviceUnavailable("Subscription request timed out"))
+
+  return makeRestResponse(req.requestId, subFut.read())
+
+proc installFilterPostSubscriptionsHandler*(router: var RestRouter,
+                                              node: WakuNode) =
+  router.api(MethodPost, ROUTE_FILTER_SUBSCRIPTIONS) do (contentBody: Option[ContentBody]) -> RestApiResponse:
+    ## Subscribes a node to a list of contentTopics of a pubsubTopic
+    debug "post", ROUTE_FILTER_SUBSCRIPTIONS, contentBody
+
+    let response = await filterPostPutSubscriptionRequestHandler(node, contentBody)
+    return response
 
-    # Successfully subscribed to all content filters
-    for cTopic in req.contentFilters:
-      cache.subscribe(cTopic)
+proc installFilterPutSubscriptionsHandler*(router: var RestRouter,
+                                           node: WakuNode) =
+  router.api(MethodPut, ROUTE_FILTER_SUBSCRIPTIONS) do (contentBody: Option[ContentBody]) -> RestApiResponse:
+    ## Modifies a subscribtion of a node to a list of contentTopics of a pubsubTopic
+    debug "put", ROUTE_FILTER_SUBSCRIPTIONS, contentBody
 
-    return RestApiResponse.ok()
+    let response = await filterPostPutSubscriptionRequestHandler(node, contentBody)
+    return response
 
-proc installFilterDeleteSubscriptionsV1Handler*(router: var RestRouter, 
-                                                node: WakuNode, 
-                                                cache: MessageCache) =
-  router.api(MethodDelete, ROUTE_FILTER_SUBSCRIPTIONSV1) do (contentBody: Option[ContentBody]) -> RestApiResponse:
+proc installFilterDeleteSubscriptionsHandler*(router: var RestRouter,
+                                                node: WakuNode) =
+  router.api(MethodDelete, ROUTE_FILTER_SUBSCRIPTIONS) do (contentBody: Option[ContentBody]) -> RestApiResponse:
     ## Subscribes a node to a list of contentTopics of a PubSub topic
-    # debug "delete_waku_v2_filter_v1_subscriptions"
+    debug "delete", ROUTE_FILTER_SUBSCRIPTIONS, contentBody
 
-    let decodedBody = decodeRequestBody[FilterSubscriptionsRequest](contentBody)
+    let decodedBody = decodeRequestBody[FilterUnsubscribeRequest](contentBody)
 
     if decodedBody.isErr():
-      return decodedBody.error
+      return makeRestResponse("unknown", FilterSubscribeError.badRequest("Failed to decode request"))
+
+    let req: FilterUnsubscribeRequest = decodedBody.value()
+
+    let peerOpt = node.peerManager.selectPeer(WakuFilterCodec)
```

Comment:
Doesn't this select a random peer?

When unsubscribing we should use the same peer than for subscribing no?